bioio_ome_zarr/writers/ome_zarr_writer.py

```diff
@@ -27,28 +30,35 @@ def _normalize_levelwise(
     label: str,
 ) -> List[Tuple[int, ...]]:
     """
-    Convert a single N-dim tuple or a per-level tuple-of-tuples into a
-    per-level List[Tuple[int, ...]] with validation.
+    Convert a single N-dim iterable (applied to all levels) or a per-level iterable
+    of N-dim iterables into a per-level List[Tuple[int, ...]] with validation.
+    Accepts lists, tuples, or generators.
     """
-    if len(spec) == 0:
+    top: List[Any] = list(cast(Iterable[Any], spec))
```

Comment:
what does `top` mean here? it used to be called spec but the casted List version is called top?

Reply:
also the use of `Any` here feels somehow sketchy. I don't know a better way to do it off the top of my head yet... can you just ask if the length is 0?  This is where Sequence is better than Iterable, it might make this whole function cleaner.

Reply:
updated to sequence in [76c319f](https://github.com/bioio-devs/bioio-ome-zarr/pull/87/commits/76c319fa8f9255839f72c328391ef846b45ab21a)